### PR TITLE
chore(deps): add webpack-cli to peers

### PR DESCRIPTION
**Pull request by eps1lon**
Follow-up on #2188 according to https://github.com/webpack/webpack-dev-server/pull/2188#issuecomment-520373076

- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Can publish a package with this PR and see if it actually works but these kind of fixes have been done multiple times to make yarn v2 happy. Considering `webpack-dev-server` throws at runtime anyway if `webpack-cli` is missing this shouldn't bother anyone

### Motivation / Use-Case

yarn v2 throws currently with 

```
Error: A package is trying to access another package without the second one being listed as a dependency of the first one

Required package: webpack-cli (via "webpack-cli/bin/config-yargs")
Required by: webpack-dev-server@virtual:836be30d0802b4899b9a78ca9d744f43f038cccf96d6b8307cbd938d17151f1ac108d2e64290515733c51d0949e0c6d87f9a7c9e245dfe628e5c4ef98f22d752#npm:3.8.0 
```

### Breaking Changes

I guess this new warning might be annoying if you don't run the `webpack-dev-server` cli. If this is a blocker I can try and see if https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-optional-peer-dependencies.md works for yarn v2.

### Additional Info
https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-optional-peer-dependencies.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -114,7 +114,13 @@
     "webpack-cli": "^3.3.6"
   },
   "peerDependencies": {
-    "webpack": "^4.0.0"
+    "webpack": "^4.0.0",
+    "webpack-cli": "^3.0.0"
```
Comment (evilebottnawi) [inline package.json]: Please remove `webpack-cli` from here, this line break CFA, vue-cli, angular-cli and etc big boilerplates

**Comment by arcanis** [comment]
> /cc @arcanis it is require what `webpack-cli` was in `peerDependencies`?

Yes. The `peerDependenciesMeta` field affect how package managers will treat the entries for the existing `peerDependencies` field - it doesn't extend it.

> Because we can't do this right now

Again, if you can't only because it would cause warnings to appear for npm's users then I'd say it's too bad because omitting the peer dependencies will cause both npm and Yarn to silently generate invalid dependency trees, whereas warnings will only cause a slight temporary discomfort at best.

Does it *actually* break something? As in, a consumer stops working altogether?

**Comment by evilebottnawi** [comment]
@arcanis do you know other developers, they starts create a lot of issues and send PRs with removing this, we want remove bin for webpack-dev-server@4 so i think better avoid this right now for webpack-dev-server@3

**Comment by arcanis** [comment]
> If this is a blocker I can try and see if https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-optional-peer-dependencies.md works for yarn v2.

Optional peer dependencies work for Yarn 1, Yarn 2, pnpm 3.2+, and (planned for) npm 6.11.

> you know other developers, they starts create a lot of issues and send PRs with removing this

I sympathise, however note that the same is also true for us - we get issues telling us that Yarn is installing dependencies the wrong way etc, and after digging we notice it's caused by undeclared dependencies like this. Plus I guess you still get PRs to add it, in the end 😄

**Comment by evilebottnawi** [comment]
@arcanis but here not `Optional peer dependencies`

**Comment by arcanis** [comment]
They are; that's how optional peer dependencies work.

The `peerDependenciesMeta` lists settings that modify the default behaviour for the dependencies field of the given name (`optional` means that the peer dependency being unmet shouldn't be reported as a warning, for example), but they still have to be declared in `peerDependencies` (otherwise we don't have the range hint for the peer dependency, for example).

**Comment by evilebottnawi** [comment]
@arcanis thansk for clarify, maybe we should wait https://github.com/npm/cli/pull/224 before merge and release this

**Comment by eps1lon** [comment]
@evilebottnawi `npm@6.11` was released which includes https://github.com/npm/cli/pull/224

I don't know how to make the checks pass though. Failures seem unrelated.

**Comment by hiroppy** [comment]
We need to wait for npm to be updated in Node.js.

**Comment by salper** [comment]
> We need to wait for npm to be updated in Node.js.

It seems to be the case. Any news on this?

**Comment by arcanis** [comment]
@evilebottnawi @hiroppy I've opened another PR (https://github.com/webpack/webpack-dev-server/pull/2396) that uses a smoother pattern and doesn't print any warnings even on older package managers. Can one of you give it a look and tell me how it looks? We plan to release Yarn 2 next week, it would be nice to sort this out before then if possible 😀

**Comment by jacklaurencegaray** [comment]
https://github.com/yarnpkg/berry/issues/815

**Comment by evilebottnawi** [comment]
Fixed for yarn v2, feel free to feedback

**Pull request closed without merging**